dpi/src/lib.rs

```diff
@@ -759,6 +759,120 @@ impl<P: Pixel> From<LogicalPosition<P>> for Position {
     }
 }
 
+/// A rectangle represented in logical pixels.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Default, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct LogicalRect<P> {
+    pub origin: LogicalPosition<P>,
+    pub size: LogicalSize<P>,
+}
+
+impl<P> LogicalRect<P> {
+    #[inline]
```

Comment:
I think we generally don't add inline annotations now unless it's `inline(always)`, but I don't really mind that for now.